Kerberos.NET/Cache/TicketCacheBase.cs

```diff
@@ -158,7 +158,10 @@ protected virtual async Task BackgroundCacheOperation()
 
                 if (cacheEntry != null && !cacheEntry.IsExpired(this.Configuration.Defaults.ClockSkew))
                 {
-                    await (this.Refresh?.Invoke(cacheEntry)).ConfigureAwait(false);
+                    if (this.Refresh != null)
+                    {
+                        await this.Refresh.Invoke(cacheEntry).ConfigureAwait(false);
```

Comment:
This actually regresses to a slightly more problematic issue because `Refresh` can be nulled out between the null check and the call across threads, whereas `this.Refresh?.Invoke` ends up being an atomic operation. As such, I think the fix is actually simpler: make it `await (this.Refresh?.Invoke(cacheEntry))?.ConfigureAwait(false);`

Reply:
Build is complaining:

> Kerberos.NET\Cache\TicketCacheBase.cs(161,21): Error CS1061: 'ConfiguredTaskAwaitable?' does not contain a definition for 'GetAwaiter' and no accessible extension method 'GetAwaiter' accepting a first argument of type 'ConfiguredTaskAwaitable?' could be found (are you missing a using directive or an assembly reference?)

Reply:
Hah, I suppose that's what I get for editing from my phone! I should be able to take a closer look on Friday